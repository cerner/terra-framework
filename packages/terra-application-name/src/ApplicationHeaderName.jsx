import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';

import styles from './ApplicationHeaderName.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The accessory element to be displayed next to the title.
   * */
  accessory: PropTypes.element,
  /**
   * The title branding of the application name.
   * */
  title: PropTypes.node,
};

const ApplicationHeaderName = ({
  accessory,
  title,
  ...customProps
}) => {
  const ApplicationHeaderNameClassNames = cx([
    'application-header-name',
    customProps.className,
  ]);

  return (
    <div {...customProps} className={ApplicationHeaderNameClassNames}>
      {accessory && <div className={cx('accessory')}>{accessory}</div>}
      {title && <div className={cx('title')}>{title}</div>}
    </div>
  );
};

ApplicationHeaderName.propTypes = propTypes;

export default ApplicationHeaderName;
