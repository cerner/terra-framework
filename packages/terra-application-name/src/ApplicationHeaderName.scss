/* stylelint-disable selector-class-pattern */
:local {
  .application-header-name {
    align-items: var(--terra-application-header-name-align-items, center);
    color: var(--terra-application-header-name-color, #000);
    display: flex;
    flex: var(--terra-application-header-name-flex, 0 0 auto);
  }

  .accessory {
    padding-left: var(--terra-application-header-name-accessory-padding-left, 0.3571rem);
  }

  .title {
    font-weight: var(--terra-application-header-name-title-font-weight, normal);
    overflow: var(--terra-application-header-name-title-overflow, hidden);
    padding-left: var(--terra-application-header-name-title-padding-left, 0.3571rem);
    padding-right: var(--terra-application-header-name-title-padding-right, 4rem);
    text-overflow: var(--terra-application-header-name-text-overflow, ellipsis);
    white-space: var(--terra-application-header-name-white-space, nowrap);
    word-break: var(--terra-application-header-name-word-break, normal);
    word-wrap: var(--terra-application-header-name-word-wrap, normal);
  }
}
