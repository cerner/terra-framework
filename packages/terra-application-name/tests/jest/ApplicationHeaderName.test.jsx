import React from 'react';
import ApplicationHeaderName from '../../src/ApplicationHeaderName';

describe('ApplicationHeaderName', () => {
  it('should render a default component', () => {
    const wrapper = shallow(<ApplicationHeaderName />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided accessory', () => {
    const wrapper = shallow(<ApplicationHeaderName accessory={<img alt="this is the accessory" />} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided title', () => {
    const wrapper = shallow(<ApplicationHeaderName title="title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with provided accessory and title', () => {
    const wrapper = shallow(<ApplicationHeaderName accessory={<img alt="this is the accessory" />} title="title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the class application-header-name', () => {
    const wrapper = shallow(<ApplicationHeaderName />);
    expect(wrapper.prop('className')).toContain('application-header-name');
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'clinical-lowlight-theme',
      });
    const wrapper = shallow(<ApplicationHeaderName accessory={<img alt="this is the accessory" />} title="title" />);
    expect(wrapper).toMatchSnapshot();
  });
});
