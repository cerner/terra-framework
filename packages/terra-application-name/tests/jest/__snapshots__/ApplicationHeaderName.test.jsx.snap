// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationHeaderName correctly applies the theme context className 1`] = `
<div
  className="application-header-name clinical-lowlight-theme"
>
  <div
    className="accessory"
  >
    <img
      alt="this is the accessory"
    />
  </div>
  <div
    className="title"
  >
    title
  </div>
</div>
`;

exports[`ApplicationHeaderName should render a default component 1`] = `
<div
  className="application-header-name"
/>
`;

exports[`ApplicationHeaderName should render with provided accessory 1`] = `
<div
  className="application-header-name"
>
  <div
    className="accessory"
  >
    <img
      alt="this is the accessory"
    />
  </div>
</div>
`;

exports[`ApplicationHeaderName should render with provided accessory and title 1`] = `
<div
  className="application-header-name"
>
  <div
    className="accessory"
  >
    <img
      alt="this is the accessory"
    />
  </div>
  <div
    className="title"
  >
    title
  </div>
</div>
`;

exports[`ApplicationHeaderName should render with provided title 1`] = `
<div
  className="application-header-name"
>
  <div
    className="title"
  >
    title
  </div>
</div>
`;
